pkg/cvo/metrics.go

```diff
@@ -449,3 +534,87 @@ func mostRecentTimestamp(cv *configv1.ClusterVersion) int64 {
 	}
 	return latest.Unix()
 }
+
+// Determine if the certificates have changed and need to be updated.
+// Returns true if both files have changed AND neither is an empty file.
+func certsChanged(origCertChecksum []byte, origKeyChecksum []byte, certFile, keyFile string) bool {
+	// Check if both files exist.
+	certNotEmpty, err := fileExistsAndNotEmpty(certFile)
+	if err != nil {
+		klog.Warningf("Error checking if changed TLS cert file empty/exists: %w", err)
+		return false
```

Comment:
We may not try again until the next fsnotify event, so I think we might want `bool, error` out of this function, and then:

```go
return false, err
```

here and:

```go
if event.Op != fsnotify.Chmod && event.Op != fsnotify.Remove {
  if changed, err := certsChanged(origCertChecksum, origKeyChecksum, certFile, keyFile); err != nil {
    return fmt.Errorf("checking TLS configuration: %w", err)
  } else changed {
    ...
  }
}
```

in the caller.  Or something like that...

Reply:
Here again I wasn't sure we wanted to shutdown CVO for these types of errors. If an admin improperly rotates the certs and ends up with no key file (if that's even possible?) do we want to kill CVO or just let it continue waiting for correct certs? Isn't it just going to restart and do that anyway?

Reply:
I've added crashlooping motivation over [here][1].  If that makes sense, I think we do want the fatal-to-the-CVO `return fmt.Errorf(...)` case here.

[1]: https://github.com/openshift/cluster-version-operator/pull/726#discussion_r786276935

Reply:
Restructured so this results in a graceful shutdown.

Reply:
As of 0e6136c727b05509, `certsChanged` and `fileExistsAndNotEmpty` calls both return errors which are treated as fatal.